datadog_lambda/tracing.py

```diff
@@ -89,18 +88,73 @@ def _context_obj_to_headers(obj):
     }
 
 
+def add_dd_dummy_metadata_subsegment(subsegment_metadata_key, dd_context):
+    """
+    Save the context to an X-Ray subsegment's metadata field, so the X-Ray
+    trace can be converted to a Datadog trace in the Datadog backend with
+    the correct context.
+    """
+    subsegment = xray_recorder.current_subsegment()
+    subsegment.put_metadata(
+        subsegment_metadata_key, dd_context, XraySubsegment.NAMESPACE
+    )
+
+
+def update_dd_dummy_metadata_subsegment(subsegment_metadata_key, lambda_function_tags):
+    """
+    Update the specified Datadog subsegment key's field with additional tags
+    """
+    subsegment = xray_recorder.current_subsegment()
+    if not lambda_function_tags or subsegment is None:
+        return
+    dd_subsegment = subsegment.metadata[XraySubsegment.NAMESPACE]
+    root_span_metadata = dd_subsegment.get(subsegment_metadata_key)
+    if root_span_metadata:
+        root_span_metadata.update(lambda_function_tags)
+
+
+def begin_dd_dummy_metadata_subsegment(
+    dd_context, trace_context_source, lambda_function_tags
+):
+    """
+    Create a Datadog subsegment to pass the Datadog trace context or tags into
+    its metadata field
+    """
+    if not dd_context and not lambda_function_tags:
+        return
+    # Start the Datadog dummy subsegment here
+    dd_dummy_subsegment = xray_recorder.begin_subsegment(XraySubsegment.NAME)
+
+    add_dd_dummy_metadata_subsegment(
+        XraySubsegment.ROOT_SPAN_METADATA_KEY, lambda_function_tags
+    )
+    if trace_context_source == TraceContextSource.EVENT:
+        add_dd_dummy_metadata_subsegment(XraySubsegment.TRACE_KEY, dd_context)
+
+    return dd_dummy_subsegment
+
+
+def end_dd_dummy_metadata_subsegment(dd_dummy_subsegment):
+    """
+    End the Datadog dummy metadata subsegment if it is the current subsegment
+    """
+    if dd_dummy_subsegment is None:
+        return
+    # Ensure we are ending the Datadog dummy subsegment
+    subsegment = xray_recorder.current_subsegment()
+    if subsegment is dd_dummy_subsegment:
+        xray_recorder.end_subsegment()
```

Comment:
Added a check for the current subsegment since I wasn't able to end the Datadog dummy subsegment directly.